schema.json

```diff
@@ -0,0 +1,488 @@
+                ]
+            }
+        },
+        "structs": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "name": {
+                        "type": "string",
+                        "description": "Name of the structure"
+                    },
+                    "doc": {
+                        "type": "string"
+                    },
+                    "type": {
+                        "type": "string",
+                        "description": "Type of the structure",
+                        "enum": [
+                            "base_in",
+                            "base_out",
+                            "extension_in",
+                            "extension_out",
+                            "standalone"
+                        ]
+                    },
+                    "extends": {
+                        "type": "array",
+                        "description": "Optional property, list of names of the structs that this extension structure extends",
+                        "items": {
+                            "type": "string"
+                        }
+                    },
+                    "free_members": {
+                        "type": "boolean",
+                        "description": "Optional property, to indicate if a free members function be emitted for the struct"
+                    },
+                    "members": {
+                        "type": "array",
+                        "description": "Optional property, list of struct members",
+                        "items": {
+                            "type": "object",
+                            "properties": {
+                                "name": {
+                                    "type": "string",
+                                    "description": "Name of the struct member"
+                                },
+                                "doc": {
+                                    "type": "string"
+                                },
+                                "type": {
+                                    "type": "string",
+                                    "description": "Type of the struct member"
+                                },
+                                "pointer": {
+                                    "type": "string",
+                                    "description": "Optional property, specifies if a struct member type is a pointer",
+                                    "enum": [
+                                        "immutable",
+                                        "mutable"
+                                    ]
+                                },
+                                "optional": {
+                                    "type": "boolean",
+                                    "description": "Optional property, to indicate if a struct member is optional"
```

Comment:
(Can be future work)
For the INIT macros (and generating C++ bindings), some optional members will need to know what their default value is (like WGPU_DEPTH_SLICE_UNDEFINED). Probably we can just put that here, though an alternative could be to move that information into the "type system" and give the `depthSlice` member a special type that knows its default.

Reply:
default values are planned - In-order to keep this initial PR minimal, a future PR which includes both the header generation and default values populated in the yaml at once would be better.